Source/Core/Common/MinizipUtil.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <algorithm>
+
+#include <minizip/unzip.h>
+
+#include "Common/CommonTypes.h"
+#include "Common/ScopeGuard.h"
+
+namespace Common
+{
+// Since minzip doesn't provide a way to unzip a file of a length > 65535,
+// we have to implement this ourselves
+template <typename ContiguousContainer>
+static bool ReadCurrentFileUnlimited(unzFile file, ContiguousContainer* destination)

```

Comment:
After the move to Common, this function name bears little resemblance of the function purpose. Could it be renamed to something like `DeflateZipFile`?

It should also probably be noted in the comments that the function expects the container to be able to hold the requested data before calling this function.